### DISCUSSION TITLE
Superconductivity calculations with phonons obtained from SSCHA code

### DISCUSSION THREAD

**Discussion by Kurufinve** [SSCHAcode/python-sscha#103] [Q&A] [answered]
Dear community,

I have a question regarding calculations of superconducting Tc with anharmonic phonons obtained after SSCHA relaxation. In one of the recent publications with the SSCHA code (http://www.nature.com/articles/s41586-020-1955-z), authors used the anharmonic phonons derived from the Hessian of E(R) to calculate the electron-phonon interaction and the superconducting Tc both by solving Migdal-Eliashberg equations and applying superconducting DFT. However, in the tutorials and examples provided with the SSCHA code there are no examples where the anharmonic a2F function and superconducting properties are calculated.

The question is:
How can be calculated and which way created the files with the variation of potential (dV_scf) with respect to anharmonic displacements (du_r) that are needed for calculation of electron-phonon matrix elements and subsequent calculations of superconducting properties?

It would be great if authors of the code will publish additional tutorials or examples with calculation of superconductive Tc with SSCHA phonons.

Sincerelly,
Daniil Poletaev

**Comment by mesonepigreco** [comment]
@rafbianco @mcalandra @ionerrea 
Do you have an answer? I think Daniil is right that this is a missing part of our current tutorials and example; maybe we can prepare one for the school we are planning.

**Reply by mesonepigreco** [reply]
@rafbianco 
Do you have a simple explanation of how to compute the superconductive temperature after calculating phonon with the SSCHA as it was done in all the hydrides works?

**Comment by rafbianco** [comment, marked as answer]
Hi, sorry for the late reply.
I can tell you what we have done in all the hydrides works.  It is a two step process.

1) We hacked elphon.f90 of Quantum Espresso (qe) so as to print the average of (d=dV/du)^2 on the Fermi surface at the end of the 
    electron-phonon coupling calculation done wih qe. Look for example at Eq (13) of https://arxiv.org/pdf/2102.00072.pdf, 
    without considering the  e/sqrt(M) terms. This is what we printed. 
    Notice that this first step is done through a DFPT calculation with qe, nothing to do with SSCHA. 

2) We used elph_fc.f90 of qe-5.1.0 to combine this quantity and eigenmodes and phonon frequencies taken from the dyn obtained 
     at the end of the SSCHA minimization (to be precise, elph_fc.f90  is fed with the force constant in real space 
     obtained from the dyn in reciprocal space, using q2r.f90 of qe). This allowed us to compute the Eliashberg function 
     through  Eq (12)-(13) https://arxiv.org/pdf/2102.00072.pdf, which allows to compute lambda, Tc, etc. etc.

I hope this helps

**Reply by Kurufinve** [reply]
Dear Raffaello,

Thank you very much for answer! 
So, you use the variation of Kohn-Sham potential (dvscf) from harmonic DFPT calculation and do not recalculate it with respect to anharmonic displacements, right? 
If yes, I slightly not understanding if this variation should be the same in both harmonic and anharmonic cases? Or the difference is too small so we can neglect it?
Have you in plans to publish (in your repository, for example) the hacked version of QE where the average of (d=dV/du)^2 on the Fermi surface at the end of the electron-phonon coupling calculation done with qe is printed? 
Also, I did not found the elph_fc.f90 in qe-5.1.0 release.

Sincerelly, 
Daniil Poletaev

**Reply by ionerrea** [reply]
Just a remark to Raffaello's answer, the elph_fc.f90 code is not released in the QE distribution, is a private code.

**Comment by matukumilli** [comment]
Dear SSCHA developers,

In continuation of this thread, may I know how to calculate the renormalised electronic structure corresponding to the SSCHA minimised structure. 

Regards,
Prasad Matukumilli

**Reply by mesonepigreco** [reply]
@ionerrea  @rafbianco 
Is there any plan to release the elph_fc.f90 code before the [SSCHA school](http://sscha.eu/Schools/2023/home/)? 
There should be the "Hands-on-session 7: Ion Errea: Calculation of the electron-phonon interaction and superconducting properties with the SSCHA." 
So I assume that all the material should be made public before the end of June.
